starboard/tools/testing/test_runner.py

```diff
@@ -184,6 +197,28 @@ def Kill(self):
   def Join(self):
     self.runner_thread.join()
 
+  def _Run2(self) -> None:
+    """Runs the launcher, and assigns a return code.
+    Sets return_code to the following values:
+      - 0x00-0x03F - target return code;
+      - 0x40 if return code is not available;
+      - 0x80 in case of crash;
+      - 0xС0 cannot launch the target.
+    """
+
+    status, return_code = abstract_launcher.ReturnCodeStatus.ISSUE, 0
+    try:
+      logging.info("Running launcher")
+      status, return_code = self.launcher.Run2()
+      logging.info("Finished running launcher")
+    except Exception:  # pylint: disable=broad-except
+      sys.stderr.write(f"Error while running {self.launcher.target_name}:\n")
+      traceback.print_exc(file=sys.stderr)
+
+    with self.return_code_lock:
+      self.return_code = (status << 6) | (return_code & _RETURN_CODE_MASK)
```

Comment:
As mentioned above, does `return_code` need to be. one single variable with quite non-obvious math and masking going into it ?

Reply:
See previous comment.